### PR TITLE
rgw: enable rgw compression

### PR DESCRIPTION
**Pull request by alimaredia**
Signed-off-by: Ali Maredia <amaredia@redhat.com>

I plan on adding testing for this in https://github.com/ceph/ceph-ansible/pull/1545 once https://github.com/ceph/ceph-ansible/pull/1872 is merged.

### CODE REVIEW

**Comment by alimaredia** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

File: roles/ceph-rgw/tasks/rgw_compression.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+- name: Dummy command to create default zone & zonegroup for the next command
+  command: sudo radosgw-admin --cluster={{ cluster }} user list
+  run_once: true
```
Comment (leseb) [inline roles/ceph-rgw/tasks/rgw_compression.yml]: Why sudo?

File: roles/ceph-rgw/tasks/rgw_compression.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+- name: Enable compression type for RGW
```
Comment (leseb) [inline roles/ceph-rgw/tasks/rgw_compression.yml]: No capital letters on task name

File: roles/ceph-rgw/tasks/main.yml [hunk 3]
```diff
@@ -29,6 +29,11 @@
   # Hard code this so we will skip the entire file instead of individual tasks (Default isn't Consistent)
   static: False
 
+- include: rgw_compression.yml
+  when: 
+    - rgw_compression_type is defined
```
Comment (leseb) [inline roles/ceph-rgw/tasks/main.yml]: Make it a variable and give it a default value

**Comment by alimaredia** [comment]
jenkins test luminous-ansible2.3-centos7_cluster

File: roles/ceph-rgw/tasks/main.yml [hunk 4]
```diff
@@ -11,15 +11,15 @@
   # Hard code this so we will skip the entire file instead of individual tasks (Default isn't Consistent)
   static: False
 
+- include: rgw_pool_pgs.yml
```
Comment (leseb) [inline roles/ceph-rgw/tasks/main.yml]: please name your include like this:

`- name: include rgw_pool_pgs.yml`

File: roles/ceph-rgw/tasks/main.yml [hunk 5]
```diff
@@ -29,6 +29,11 @@
   # Hard code this so we will skip the entire file instead of individual tasks (Default isn't Consistent)
   static: False
 
+- include: rgw_compression.yml
```
Comment (leseb) [inline roles/ceph-rgw/tasks/main.yml]: ditto

File: roles/ceph-rgw/tasks/main.yml [hunk 6]
```diff
@@ -29,6 +29,11 @@
...
+  when: 
+    - rgw_compression_type != "None"
```
Comment (leseb) [inline roles/ceph-rgw/tasks/main.yml]: use single quote

File: roles/ceph-rgw/tasks/rgw_compression.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+---
+- name: Dummy command to create default zone & zonegroup for the next command
```
Comment (leseb) [inline roles/ceph-rgw/tasks/rgw_compression.yml]: no capital letter in task's name

**Comment by alimaredia** [comment]
@leseb do you know why the luminous-ansible2.3-centos7_cluster tests are failing. The jenkins output tells me that it's becasue of `TASK [ceph-nfs : start nfs gateway service]`.

That task was working just fine when it was first merged a week or so ago, and I haven't done any other additional development on it.

**Comment by guits** [comment]
@alimaredia the error you are encountering with `TASK [ceph-nfs : start nfs gateway service]` is not related to your PR.
This is a SElinux error that appeared since an upgrade of vagrant centos image in our CI.

**Comment by leseb** [comment]
jenkins test this please

**Comment by leseb** [comment]
Closing this due to inactivity, feel free to re-open, thanks!

**Pull request closed without merging**